server/service/integration_core_test.go

```diff
@@ -1185,3 +1185,80 @@ func (s *integrationTestSuite) TestListActivities() {
 	require.Len(t, listResp.Activities, 1)
 	assert.Equal(t, fleet.ActivityTypeEditedPack, listResp.Activities[0].Type)
 }
+
+func (s *integrationTestSuite) TestListGetCarves() {
+	t := s.T()
+
+	ctx := context.Background()
+
+	hosts := s.createHosts(t)
+	c1, err := s.ds.NewCarve(ctx, &fleet.CarveMetadata{
+		CreatedAt: time.Now(),
+		HostId:    hosts[0].ID,
+		Name:      t.Name() + "_1",
+		SessionId: "ssn1",
+		MaxBlock:  3,
+	})
+	require.NoError(t, err)
+	c2, err := s.ds.NewCarve(ctx, &fleet.CarveMetadata{
+		CreatedAt: time.Now(),
+		HostId:    hosts[1].ID,
+		Name:      t.Name() + "_2",
+		SessionId: "ssn2",
+		MaxBlock:  3,
+	})
+	require.NoError(t, err)
+	c3, err := s.ds.NewCarve(ctx, &fleet.CarveMetadata{
+		CreatedAt: time.Now(),
+		HostId:    hosts[2].ID,
+		Name:      t.Name() + "_3",
+		SessionId: "ssn3",
+	})
+	require.NoError(t, err)
+
+	// set c1 max block
```

Comment:
Q: Is it changing max block here?

Reply:
Ah yeah, it's confusing because NewCarve does not set the max block, but because the MaxBlock field was set on the struct passed to NewCarve, calling UpdateCarve here will actually set the max block to 3. I'll change the code to make that more obvious, by moving the MaxBlock field assignment here before UpdateCarve.

Reply:
Ah makes sense now, thanks!